pkg/jaeger/store/binary_tags.go

```diff
@@ -0,0 +1,136 @@
+	return len(t.spanTags) == 0 && len(t.logsTags) == 0 && len(t.processTags) == 0
+}
+
+func isEncodedBinaryValue(v interface{}) bool {
+	if val, isStr := v.(string); isStr {
+		if strings.HasPrefix(val, VALUE_TYPE_BINARY) {
+			return true
+		}
+	}
+	return false
+}
+
+// decodeSpanBinaryTags decodes the tags with binary values that are present in
+// the binaryTags sets for the span, process and logs.
+//
+// The spanBinaryTags struct is necesary in order to not go through all the
+// tags for each Span that the query is returning. We are taking advantage on
+// the fact that when translating from OTEL to Jaeger we had to visit all the
+// tags and were able to construct this set.
+//
+// When writing binary tags we encode the slice of bytes into a base64 string
+// representation and add the prefix `__ValueType_BINARY__`. Decoding implies
+// removing the prefix and decoding the base64 string.
+func decodeSpanBinaryTags(span *model.Span, binaryTags *spanBinaryTags) {
+	if len(binaryTags.spanTags) != 0 {
+		decodeBinaryTags(span.Tags, binaryTags.spanTags)
+	}
+
+	if len(binaryTags.processTags) != 0 {
+		decodeBinaryTags(span.Process.Tags, binaryTags.processTags)
+	}
+
+	if len(binaryTags.logsTags) != 0 {
+		for i, logsTags := range binaryTags.logsTags {
+			decodeBinaryTags(span.Logs[i].Fields, logsTags)
+		}
+	}
+}
+
+func decodeBinaryTags(actualTags []model.KeyValue, binaryTagsToDecode map[string]struct{}) {
+	for i, tag := range actualTags {
+		_, ok := binaryTagsToDecode[tag.Key]
+		if !ok {
+			continue
+		}
+
+		if tag.GetVType() != model.ValueType_STRING {
+			continue
+		}
+
+		encoded := tag.VStr
+		if !strings.HasPrefix(encoded, VALUE_TYPE_BINARY) {
+			continue
+		}
+
+		vBin, err := decodeBinaryTagValue(encoded)
+
+		// If we can't decode it means that we didn't encode it in the
+		// first place, so we should keep it as is.
+		if err != nil {
+			continue
+		}
+		actualTags[i] = model.KeyValue{
+			Key:     tag.Key,
+			VType:   model.ValueType_BINARY,
+			VBinary: vBin,
+		}
+	}
+}
+
+func decodeBinaryTagValue(encoded string) ([]byte, error) {
+	v := encoded[VALUE_TYPE_BINARY_LEN:]
+	return base64.StdEncoding.DecodeString(v)
+}
+
+func encodeBinaryTagToStr(tag model.KeyValue) model.KeyValue {
+	value := fmt.Sprintf("__ValueType_BINARY__%s", base64.StdEncoding.EncodeToString(tag.GetVBinary()))
+	return model.KeyValue{
+		Key:   tag.Key,
+		VType: model.ValueType_STRING,
+		VStr:  value,
+	}
+}
+
+func encodeBinaryTags(span *model.Span) {
+	for i, tag := range span.Tags {
+		if isBinaryTag(tag) {
+			continue
+		}
+		span.Tags[i] = encodeBinaryTagToStr(tag)
+	}
+
+	for _, log := range span.Logs {
+		for i, tag := range log.Fields {
+			if isBinaryTag(tag) {
+				continue
+			}
+			log.Fields[i] = encodeBinaryTagToStr(tag)
+		}
+	}
+
+	for i, tag := range span.Process.Tags {
+		if isBinaryTag(tag) {
+			continue
+		}
+		span.Process.Tags[i] = encodeBinaryTagToStr(tag)
+	}
+}
+
+func isBinaryTag(tag model.KeyValue) bool {
```

Comment:
This function does exactly opposite of what the name tells. Either fix the name or `if `handler

Reply:
OMG xD